src/util/framerate.ts

```diff
@@ -296,9 +299,23 @@ export class DownsamplingBasedOnFrameRateCalculator {
       Math.pow(2, Math.round(Math.log2(downsampleFactorBasedOnFramerate))),
       this.maxDownsamplingFactor,
     );
-    return this.updateMaxTrackedDownsamplingRate(
+    const applied = this.updateMaxTrackedDownsamplingRate(
       downsampleFactorBasedOnFramerate,
     );
+    if (this.enableLogging) {
+      // Only log when the applied factor actually changes compared to previous frame.
+      if ((this as any)._lastLoggedFactor !== applied) {
```

Comment:
Perhaps `lastLoggedFactor` could be a regular class attribute to avoid the type cast to any

Reply:
agreed, fixed.